### PR TITLE
Suppress standard emails for POS-paid orders and auto-trigger POS emails

### PR DESCRIPTION
**Pull request by staskus**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

Related PR: https://github.com/Automattic/woocommerce-payments/pull/11362

WOOMOB-2087
WOOMOB-2161

### Changes proposed in this Pull Request:

This is part of the POS bookings support initiative: suppressing standard WooCommerce emails for orders paid at POS and enabling POS-specific email templates that fire both automatically and manually.

**What this PR does:**

1. **Adds `is_order_paid_at_pos()` to `PointOfSaleOrderUtil`**: Detects whether an order was paid at POS regardless of how it was created. An order is considered paid at POS if:
   - It was created via the POS REST API (`created_via = 'pos-rest-api'`), OR
   - It was paid via card terminal (`_wcpay_ipp_channel = mobile_pos` meta present: set by WooCommerce Payments), OR
   - It was paid via cash at POS (`_cash_change_amount` meta present).

2. **Adds `PointOfSaleEmailHandler`**: A new service that hooks into WooCommerce email filters to suppress standard customer and admin emails (completed, processing, on-hold, refunded, new order) for orders paid at POS. This allows the POS app to control its own email/receipt flow instead.

3. **POS emails are both automatic and manual**: The POS completed and refunded email templates:
   - **Automatically fire** when a POS-paid order is completed (`auto_trigger` on `woocommerce_order_status_completed_notification`) or refunded (`auto_trigger` on `woocommerce_order_fully_refunded_notification` / `woocommerce_order_partially_refunded_notification`).
   - **Can also be manually sent** via the REST API "Email receipt" button (`trigger` on `woocommerce_rest_order_actions_email_send`).
   - The `auto_trigger()` methods use the `is_order_paid_at_pos()` guard, so they only fire for POS orders.
   - The POS completed email template is eligible for any POS-paid order regardless of current status (e.g., processing, on-hold, completed), so receipts can be resent at any time.

4. **POS refunded email uses the specific refund**: The `trigger()` method accepts the `refund_id` from the notification hook and resolves that specific refund instead of picking the most recent one via `get_refunds()`. This ensures the correct refund details are included when multiple refunds exist on an order. When resent via the REST API (no `refund_id`), refund details are omitted.

5. **Uses the `_wcpay_ipp_channel` meta key**: This checks for `_wcpay_ipp_channel = mobile_pos`, which is set by WooCommerce Payments for POS card terminal payments via the In-Person Payments (IPP) channel.

### How to test the changes in this Pull Request:

1. Enable the POS feature flag: `wp option update woocommerce_feature_point_of_sale_enabled yes`
2. Install and configure CIAB Store with this version of WooCommerce and this version of Woo Payments https://github.com/Automattic/woocommerce-payments/pull/11362 or use pre-configured https://totally-very-arcade.commerce-garden.com/ site
3. Create a booking order (this creates an order with `created_via = 'bookings'`).
4. Pay for the booking at POS using the card reader.
5. Verify that standard WooCommerce emails (completed order, processing order, new order) are **not** sent for this order.
6. Verify that the **POS completed order email** is automatically sent when the order status is set to completed.
7. Send a manual email receipt and verify it is sent.
8. Verify that the POS completed email receipt can be resent from a POS-paid order in any status (e.g., processing, on-hold), not only completed.
9. Create a refund on a POS-paid order and verify the **POS refunded order email** is automatically sent with the correct refund details.
10. Create multiple refunds on the same order and verify each refund email references the specific refund that triggered it.
11. Create a regular web order through the checkout and verify that standard emails **are** still sent normally.
12. Create a POS order (`created_via = 'pos-rest-api'`) and verify standard emails are suppressed but POS emails fire.
13. Pay an order with cash at POS (has `_cash_change_amount` meta) and verify standard emails are suppressed but POS emails fire.

image

only a single POS email should be sent after the payment on POS.

### Testing that has already taken place:

- PHPUnit tests cover all scenarios for `PointOfSaleOrderUtil::is_order_paid_at_pos()` (POS-created, card terminal, cash, zero change, regular web, bookings without POS).
- PHPUnit tests cover `PointOfSaleEmailHandler::maybe_suppress_email()` for all order types and edge cases (null order, non-order object, already disabled emails).
- PHPUnit tests verify filter registration behavior based on POS feature flag state.
- PHPUnit tests verify that the refunded email trigger uses the specific refund_id when provided and handles the no-refund-id resend path.
- PHPStan analysis passes with no errors.
- PHPCS linting passes with no errors.

### Milestone

- [x] Automatically assign milestone for the **[next WooCommerce version](../blob/trunk/plugins/woocommerce/woocommerce.php#L6)**

> **Note:** Check the box above to have the milestone automatically assigned when merged.
> Alternatively (e.g. for point releases), manually assign the appropriate milestone.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Comment</summary>

#### Comment

Internal feature behind POS feature flag - no user-facing changes for non-POS users.

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/emails/class-wc-email-customer-pos-refunded-order.php [hunk 1]
```diff
@@ -164,60 +165,58 @@ public function get_heading() {
...
+		 * @param int $order_id  The order ID.
+		 * @param int $refund_id The refund ID.
 		 *
-		 * @internal For exclusive usage within this class, backwards compatibility not guaranteed.
+		 * @internal
+		 * @since 10.6.0
 		 */
-		public function trigger_full( $order_id, $refund_id = null ) {
-			$this->trigger( $order_id, false, $refund_id );
+		public function auto_trigger( $order_id, $refund_id = null ): void {
```
Comment (toupper) [inline plugins/woocommerce/includes/emails/class-wc-email-customer-pos-refunded-order.php:176]: If I see this correctly, `auto_trigger()` receives `$refund_id` but it is never used. This loses the exact refund context that fired the hook (woocommerce_order_*_refunded_notification). Perhaps we can pass `$refund_id` through to `trigger()` and set `$this->refund = wc_get_order( $refund_id )` to mirror core refunded email behavior?

File: plugins/woocommerce/includes/emails/class-wc-email-customer-pos-refunded-order.php [hunk 2]
```diff
@@ -164,60 +165,58 @@ public function get_heading() {
...
-			if ( ! empty( $refund_id ) ) {
-				$this->refund = wc_get_order( $refund_id );
-			} else {
-				$this->refund = false;
-			}
+			$this->placeholders['{order_date}']   = wc_format_datetime( $order->get_date_created() );
+			$this->placeholders['{order_number}'] = $order->get_order_number();
+
+			$refunds      = $order->get_refunds();
+			$this->refund = ! empty( $refunds ) ? reset( $refunds ) : false;
```
Comment (toupper) [inline plugins/woocommerce/includes/emails/class-wc-email-customer-pos-refunded-order.php]: If I follow this line correctly, `$this->refund = ! empty( $refunds ) ? reset( $refunds ) : false; `can select an arbitrary refund when an order has multiple refunds, right? If that it's so, that can mismatch the refund event that triggered this email. Can we use the specific `$refund_id `from the action path instead of re-selecting from `get_refunds()`?

File: plugins/woocommerce/tests/php/src/Internal/Orders/PointOfSaleEmailHandlerTest.php [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+	 * @testdox maybe_suppress_email returns original value when object is not an order.
+	 */
+	public function test_returns_original_value_for_non_order_object(): void {
+		$this->assertTrue( $this->sut->maybe_suppress_email( true, new \stdClass() ) );
+	}
+
+	/**
+	 * @testdox register adds filters only when POS feature is enabled.
+	 */
+	public function test_register_adds_filters_when_pos_enabled(): void {
```
Comment (toupper) [inline plugins/woocommerce/tests/php/src/Internal/Orders/PointOfSaleEmailHandlerTest.php:106]: Coverage currently validates suppression and hook registration, but not `refund-email` refund context, right? Shall we add a test proving POS refunded emails use the triggering `$refund_id `(especially when an order has multiple refunds), to cover the comment above?

**Review by toupper — COMMENTED** [review]
@staskus I didn't have the chance to test it as it failed, but here I add a few comments in the code.

File: plugins/woocommerce/includes/emails/class-wc-email-customer-pos-refunded-order.php (hunk 2 above)
Comment (staskus) [reply plugins/woocommerce/includes/emails/class-wc-email-customer-pos-refunded-order.php]: That's a good comment. Yes, it can be problematic 🤔 I'll address it.

File: plugins/woocommerce/tests/php/includes/emails/class-wc-email-customer-pos-refunded-order-test.php [hunk 4]
```diff
@@ -422,4 +422,59 @@ public function test_email_content_does_not_include_refund_returns_policy_when_o
...
+			)
+		);
+
+		$email = new WC_Email_Customer_POS_Refunded_Order();
+
+		// Trigger with the first refund's ID.
+		$email->trigger( $order->get_id(), $email->id, $first_refund->get_id() );
+
+		$this->assertSame( $first_refund->get_id(), $email->refund->get_id(), 'Should use the specific refund that triggered the email' );
+	}
```
[the remark this answers is not in the document]
Comment (staskus) [reply plugins/woocommerce/tests/php/includes/emails/class-wc-email-customer-pos-refunded-order-test.php:456]: No other tests in this file guard against WP_Error either

**Comment by staskus** [comment]
Hey, @Konamiman. 

Would you mind taking a look at this PR when you get a chance? This is a more expanded version [of the previous PR](https://github.com/woocommerce/woocommerce/pull/62976), containing the changes we need for the POS CIAB Bookings support.

Our team is doing the testing and review as well, but we need a more experienced sanity check on the code changes.

Here's a quick summary of what it does:

#### Core addition: `PointOfSaleOrderUtil::is_order_paid_at_pos()`
 
A new utility method that detects whether an order was paid at POS, regardless of how it was created (e.g., bookings). It checks for POS REST API origin, card terminal payment (_wcpay_ipp_channel), or cash payment (_cash_change_amount).

#### Email suppression: `PointOfSaleEmailHandler`

A new service that hooks into WooCommerce email filters to suppress standard customer and admin emails (completed, processing, on-hold, refunded, new order) for POS-paid orders. This lets the POS app control its own receipt flow.

#### POS-specific emails (completed + refunded)
  - Auto-trigger when order status changes (completed/refunded), gated by the `is_order_paid_at_pos` check
  - Manually send via `orders/send_email` 

 Let me know! Thanks.

**Review by samiuelson — APPROVED** [review]
I've tested E2E receipt email flow in Woo POS mobile app on Android—it works as expected 👍

**Review by Konamiman — APPROVED** [review]

**Comment by tpaksu** [comment]
Hi, after pulling the latest trunk, it started to give me the infamous `load_textdomain_just_in_time` error:

```
Notice: Function _load_textdomain_just_in_time was called <strong>incorrectly</strong>. Translation loading for the <code>woocommerce</code> domain was triggered to
     o early. This is usually an indicator for some code in the plugin or theme running too early. Translations should be loaded at the <code>init</code> action or later
     . Please see <a href="https://developer.wordpress.org/advanced-administration/debug/debug-wordpress/">Debugging in WordPress</a> for more information.
```

and Claude found that it's being thrown because: 

> PointOfSaleEmailHandler::register() was called during WooCommerce::__construct() → init_hooks() (plugin loading phase, before WordPress init action). It called FeaturesUtil::feature_is_enabled(), which triggered FeaturesController::init_feature_definitions(), which uses __() for translating feature names. This loaded the woocommerce text domain before init, triggering the WordPress 6.7+ notice.

And it suggested this fix:

```php 
    /**
	 * Register hooks and filters.
	 */
	public function register(): void {
		// Defer feature check to 'init' to avoid triggering translation loading too early (WordPress 6.7+).
		add_action(
			'init',
			function () {
				if ( ! FeaturesUtil::feature_is_enabled( 'point_of_sale' ) ) {
					return;
				}

				foreach ( self::SUPPRESSED_EMAIL_IDS as $email_id ) {
					add_filter( 'woocommerce_email_enabled_' . $email_id, array( $this, 'maybe_suppress_email' ), 10, 2 );
				}
			}
		);
	}
```

For your information :)

**Comment by staskus** [comment]
@tpaksu thanks for flagging,. on it!

**Comment by staskus** [comment]
Fix: https://github.com/woocommerce/woocommerce/pull/63441
